ncm-network: should test for /etc/noquattor first and return a warning, not an error

If `/etc/noquattor` exists when the `ncm-network` component is run, we get:

```
[INFO] running component: network
---------------------------------------------------------
[INFO] test_network_ccm_fetch: trying ccm-fetch
[ERROR] Error output: [WARN] CCM updates disabled globally (/etc/noquattor present)

[WARN] test_network_ccm_fetch: FAILED: network down
[ERROR] Network restart failed. Reverting back to original config. Failed modified
configfiles can be found in /tmp/ with suffix -failed. (If there aren't any, it means only
some devices were removed.)
[INFO] test_network_ccm_fetch: trying ccm-fetch
[ERROR] Error output: [WARN] CCM updates disabled globally (/etc/noquattor present)

[WARN] test_network_ccm_fetch: FAILED: network down
[ERROR] Restoring old config failed.

<snip to remove pages of debug info>

[INFO] The profile of this machine could not be retrieved using standard mechanism
ccm-fetch. Since this should be the original configuration, there's either a bug in
ncm-network or your profile server is/was not reachable. Run "ccm-fetch" and then
"ncm-ncd --co network" to find out more. If you think there's a bug in this component,
please let us know.
[INFO] configure on component network executed, 4 errors, 2 warnings
```

Note in particular the line:

```
[ERROR] Error output: [WARN] CCM updates disabled globally (/etc/noquattor present)
```

So it is running `ccm-fetch`, which at [line 203](https://github.com/quattor/CCM/blob/master/src/main/bin/ccm-fetch#L202) returns a _warning_ that `/etc/noquattor` exists.

Unfortunately, `ncm-network` at [line 838](https://github.com/quattor/configuration-modules-core/blob/master/ncm-network/src/main/perl/network.pm#L831) interprets _any_ non-zero exit code from `ccm-fetch` as an error, not a warning, and goes on to report `FAILED: network down`.

Can the presence of `/etc/noquattor` remain as a warning please?  Can `ncm-network` test for the presence of this file before doing anything else, and return a warning if it exists without attempting to do anything else?

--- a/ncm-network/src/main/perl/network.pm
+++ b/ncm-network/src/main/perl/network.pm
@@ -47,6 +47,7 @@ use CAF::FileEditor;
 use CAF::FileWriter;
 use Fcntl qw(SEEK_SET);
 use LC::File;
+use POSIX qw(WIFEXITED WEXITSTATUS);
 
 # Ethtool formats query information differently from set parameters so
 # we have to convert the queries to see if the value is already set correctly
@@ -833,11 +834,11 @@ sub Configure {
         # no runrun, as it would trigger error (and dependency failure)
         my $output = CAF::Process->new(["ccm-fetch"], log => $self)->output();
         my $exitcode=$?;
-        if ($? == 0) {
+        if ($exitcode == 0) {
             $self->info("$func: OK: network up");
             return 1;
-        } elsif (($?>>8) == NOQUATTOR_EXITCODE) {
-            $self->warn("$func: ccm-fetch failed due with NOQUATTOR. Testing network with ping.");
+        } elsif (WIFEXITED($exitcode) && WEXITSTATUS($exitcode) == NOQUATTOR_EXITCODE) {
+            $self->warn("$func: ccm-fetch failed with NOQUATTOR. Testing network with ping.");
             return test_network_ping();
         } else {
             $self->warn("$func: FAILED: network down");

